### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by engineer027**

### CODE REVIEW

File: src/main/java/core/basesyntax/Figure.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package core.basesyntax;
+
+public abstract class Figure {
+    protected Color color;
+    protected int sideLengthA;
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Figure.java:5]: does circle have sideLengthA and B?

File: src/main/java/core/basesyntax/Perimeter.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package core.basesyntax;
+
+public interface Perimeter {
+
+    int perimeterFigure();
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Perimeter.java:5]: same here with name

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Area.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package core.basesyntax;
+
+public interface Area {
+
+    int areaFigure();
```
[the remark this answers is not in the document]
Comment (engineer027) [reply src/main/java/core/basesyntax/Area.java]: done

File: src/main/java/core/basesyntax/Circle.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    public int getRadius() {
+        return radius;
+    }
+
+    public void setRadius(int radius) {
+        this.radius = radius;
+    }
+
+    public Color getColor() {
```
[the remark this answers is not in the document]
Comment (engineer027) [reply src/main/java/core/basesyntax/Circle.java]: done

File: src/main/java/core/basesyntax/Figure.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+    protected int sideLengthB;
+
+    public abstract void printParameter();
```
[the remark this answers is not in the document]
Comment (engineer027) [reply src/main/java/core/basesyntax/Figure.java]: done

File: src/main/java/core/basesyntax/Circle.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+public class Circle extends Figure implements Area {
```
[the remark this answers is not in the document]
Comment (engineer027) [reply src/main/java/core/basesyntax/Circle.java]: done

File: src/main/java/core/basesyntax/Figure.java (hunk 1 above)
Comment (engineer027) [reply src/main/java/core/basesyntax/Figure.java]: done

File: src/main/java/core/basesyntax/Perimeter.java (hunk 2 above)
Comment (engineer027) [reply src/main/java/core/basesyntax/Perimeter.java]: done

File: src/main/java/core/basesyntax/FigureProducer.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureProducer {
+    public static final int COUNT_OF_FIGURE_IMPLEMENTATIONS = 5;
+    ColorProducer color = new ColorProducer();
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/FigureProducer.java:7]: don't forget about access modifiers, should be default

File: src/main/java/core/basesyntax/FigureProducer.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureProducer {
+    public static final int COUNT_OF_FIGURE_IMPLEMENTATIONS = 5;
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/FigureProducer.java:6]:
```suggestion
    private static final int COUNT_OF_FIGURE_IMPLEMENTATIONS = 5;
```
if we use this variable only in this class make it private

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/FigureProducer.java (hunk 8 above)
Comment (engineer027) [reply src/main/java/core/basesyntax/FigureProducer.java:6]: done

File: src/main/java/core/basesyntax/Figure.java [hunk 9]
```diff
@@ -0,0 +1,11 @@
+package core.basesyntax;
+
+public abstract class Figure implements Area, Parameter {
+    protected Color color;
+
+    public Color getColor() {
+        return color;
+    }
+
+    //public abstract void printParameter();
```
[the remark this answers is not in the document]
Comment (engineer027) [reply src/main/java/core/basesyntax/Figure.java:10]: done

File: src/main/java/core/basesyntax/FigureProducer.java (hunk 7 above)
Comment (engineer027) [reply src/main/java/core/basesyntax/FigureProducer.java]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request closed without merging**